src/briefcase/integrations/android_sdk.py

```diff
@@ -342,29 +628,40 @@ def start_app(self, package, activity):
         name, you can find it using `aapt dump badging filename.apk` and looking
         for "package" and "launchable-activity" in the output.
         """
-        # `adb shell am start` always exits with status zero. We look for error
-        # messages in the output.
-        output = self.run(
-            "shell",
-            "am",
-            "start",
-            "{package}/{activity}".format(package=package, activity=activity),
-            "-a",
-            "android.intent.action.MAIN",
-            "-c",
-            "android.intent.category.LAUNCHER",
-        ).decode("ascii", "replace")
-
-        if any(
-            (
-                line.startswith("Error: Activity class ")
-                and line.endswith("does not exist.")
-                for line in output.split("\n")
+        try:
+            # `adb shell am start` always exits with status zero. We look for error
+            # messages in the output.
+            output = self.run(
+                "shell",
+                "am",
+                "start",
+                "{package}/{activity}".format(package=package, activity=activity),
+                "-a",
+                "android.intent.action.MAIN",
+                "-c",
+                "android.intent.category.LAUNCHER",
             )
-        ):
-            raise BriefcaseCommandError("""\
+
+            if any(
+                (
+                    line.startswith("Error: Activity class ")
+                    and line.endswith("does not exist.")
+                    for line in output.split("\n")
+                )
+            ):
+                raise BriefcaseCommandError(
+                    """\
     Activity class not found while starting app.
 
     `adb` output:
 
-    {output}""".format(output=output))
+    {output}""".format(
+                        output=output
+                    )
+                )
+        except subprocess.CalledProcessError:
```

Comment:
Have you seen this error occur? If not, I worry we're not catching any realistic error condition.

The way `BriefcaseCommandError` works is it squelches any other error reporting, so IMHO the `except` clause here makes it harder to debug any unknown-unknown-type issue.

Reply:
Agreed the "unknown unknown" error handling isn't great; but we *will* get an explicit error that says what ADB command failed, so the debugging path (get them to manually re-run the command in question) should be clear.